### DISCUSSION TITLE
How to display the Instagram embed?

### DISCUSSION THREAD

**Discussion by ru88s** [vercel/next.js#16677] [Help]
I using dangerouslySetInnerHTML but it doesn't show up.
It can't be displayed even if useEffect is used.
What should I do?

```
  useEffect(() => {
    const s = document.createElement("script");
    s.setAttribute("src", "//www.instagram.com/embed.js");
    s.setAttribute("async", "true");
    document.head.appendChild(s);
  }, []);
```

**Comment by joshrobertsv2** [comment]
I had a similar issue yesterday, the solution I came up with may not be the best because you have to parse HTML to locate the script (if there is one) but at least it works. You will need Next.js <Script /> component. 

`
import Script from 'next/script';
import { useId } from 'react';
import { YourComponent } from './thisFile'

export function YourComponent ({html}) {
// this regex will parse through the html block looking for a script tag, if it finds one, the value will be the src from the script
// example: console.log(src) --->  "https://instagram.embed.js.com"
const scriptSRC = html.match(/src="([^"]*)"/)?.[1];

return ( 
<YourComponent
dangerouslySetInnerHTML={{ __html: html }}
 />
 {scriptSRC && <Script src={scriptSRC} id={id} />}
)
`
